htdocs/mrp/class/mo.class.php

```diff
@@ -1659,6 +1659,43 @@ public function getMoChilds()
 		}
 	}
 
+	/**
+	 * Function used to return all child MOs recursively
+	 *
+	 * @return Mo[]|int[]  array of MOs if OK, -1 if KO
+	 */
+	public function getAllMoChilds()
```

Comment:
Sometimes, the database is corrupted with child that has a child that is parent.
This lead to infinite loop. For this reason on Dolibarr, when we make a recursing method, we always add a counter to counter the depth and stop when depth is 1000 (just to avoid infinite loop).
Can you introduce a parameter $depth = 0 into method, and increase +1 at each recursive call and add a test to stop when value > 1000
(1000 can be hardcoded because it should not happen in common cases).